provider/openstack/image.go

```diff
@@ -44,6 +44,15 @@ func findInstanceSpec(
 	if err != nil {
 		return nil, err
 	}
+
+	if ic.Constraints.HasRootDiskSource() && *ic.Constraints.RootDiskSource == "volume" {
+		// When the root disk is a volume (i.e. cinder block volume)
+		// we don't want to match on RootDisk size. If an instance requires
+		// a very large root disk we don't want to select a larger instance type
+		// to fit a disk that won't be local to the instance.
+		ic.Constraints.RootDisk = nil
```

Comment:
Instead of just overwriting a potential user specified root disk size we need to use constraint validation to explicitly disallow root-disk-source="volume" and root-disk != nil for openstack
See InstancePrechecker inferface

Reply:
RootDisk is only set to nil for findInstanceSpec and only affects the matching of an instanceSpec based on constraints. In the case of RootDiskSource=volume, we want to change the meaning of RootDisk to the size of the cinder volume not the size of the local storage on the instance. Possibly we need to rethink this. I'll have a look at InstancePrechecker and how I can use it.